pennylane/ops/op_math/prod.py

```diff
@@ -226,6 +227,16 @@ def is_hermitian(self):
                 return False
         return all(op.is_hermitian for op in self.factors)
 
+    @property
+    def overlapping_wires(self) -> bool:
+        """Boolean expression that indicates if the factors have overlapping wires."""
+        if self._overlapping_wires is None:
+            wires = []
+            for op in self.factors:
+                wires.extend(list(op.wires))
```

Comment:
We need a list of all the factor's wires to see if there is any repeated value (thus there are overlapping wires).

This can be easily checked by converting this list into a set, and checking if the length is still the same.